plugins/speccy.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+from cloudbot import hook
+
+import re
+import requests
+
+from bs4 import BeautifulSoup as BS
+from contextlib import closing
+
+url_re = re.compile('https?:\/\/speccy.piriform.com\/results\/[A-z0-9]+', re.I)
```

Comment:
You'll want to escape those `.`, otherwise they will match any character.
I also recommend against doing `[A-z]` as that includes all characters between the upper and lowercase letters, such as `[`, `\`, `]`, `^`, `_`, and `` ` ``.